### ISSUE TITLE
Add Grants to Materializations

### ISSUE THREAD

**Issue by nathaniel-may** [dbt-labs/dbt-snowflake#168]
blocked by: dbt-labs/dbt-core#5263

Once the above work is merged into core, there are some tweaks that need to be made in the Snowflake adapter to fully expose the feature:
    - [ ] add a call to `apply_grants(..., revoke=True)` to all custom materializations
    - [ ] in view/table materializations, call apply_grants(..., revoke=False) instead for a performance improvement

**Comment by jtcohen6** [comment]
- `get_grant_sql`: https://docs.snowflake.com/en/sql-reference/sql/grant-privilege.html (I think the default should work)
- `get_revoke_sql`: https://docs.snowflake.com/en/sql-reference/sql/revoke-privilege.html (I think the default should work)
- `get_show_grants_sql`: https://docs.snowflake.com/en/sql-reference/sql/show-grants.html (different from PostgreSQL — but we might want to make this the default?)
